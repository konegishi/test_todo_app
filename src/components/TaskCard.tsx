import React from 'react';
import Task from './Task';

/**
 * TaskCardのProps
 */
interface TaskCardProps {
  /** タスクのリスト */
  todos: {
    task: string;
    id: number;
    description: string;
    is_complete: boolean;
  }[];
  /** DeleteボタンのHandler */
  deleteTodoHandler: (id: number) => void;
  /** UpdateボタンのHandler */
  updateTodoHandler: (title: string, description: string, id: number) => void;
  /** CompleteチェックボックスのHandler */
  updateCompleteFlagHandler: (isComplete: boolean, id: number) => void;
}

/**
 * TaskCardコンポーネント
 * 複数のタスクを1つにまとめるためのコンポーネント
 *
 * @param props TaskCardのProps
 * @returns TaskCardコンポーネント
 */
const TaskCard: React.FC<TaskCardProps> = (props) => {
  return (
    <React.Fragment>
      <div className='flex-grow pt-4 px-4'>
        <div className=' bg-white rounded mb-4 shadow-lg p-2'>
          <h2 className='ml-2 mb-2 font-semibold text-2xl text-blueGray-700'>
            今日
          </h2>
          {props.todos.map((todo) => {
            return (
              <Task
                title={todo.task}
                id={todo.id}
                description={todo.description}
                key={todo.id}
                isComplete={todo.is_complete}
                deleteTodoHandler={props.deleteTodoHandler}
                updateTodoHandler={props.updateTodoHandler}
                updateCompleteFlagHandler={props.updateCompleteFlagHandler}
              />
            );
          })}
        </div>
      </div>
    </React.Fragment>
  );
};

export default TaskCard;
