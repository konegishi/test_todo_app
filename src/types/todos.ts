import { User } from '@supabase/supabase-js';

/**
 * Todoの型定義
 */
export interface Todo {
  /** todo名 */
  task: string;
  /** todoのid */
  id: number;
  /** 説明 */
  description: string;
  /** 完了フラグ */
  isComplete: boolean;
}

export interface UseTodos {
  /** todoのリスト */
  todos: Todo[];
  /** ローディング中フラグ */
  isLoading: boolean;
  /**addTodoメソッドの型定義 */
  addTodo: (user: User, taskText: string) => Promise<void>;
  /** updateTodoメソッドの型定義 */
  updateTodo: (
    id: number,
    todo: {
      task?: string;
      description?: string;
      isComplete?: boolean;
    }
  ) => Promise<void>;
  /** deleteTodoメソッドの型定義 */
  deleteTodo: (id: number) => Promise<void>;
}
