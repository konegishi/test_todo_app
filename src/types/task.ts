import { UseTodos } from './todos';

/**
 * TaskコンポーネントのProps
 */
export interface TaskProps {
  /** タスク名 */
  title: string;
  /** タスクのid */
  id: number;
  /** タスクの説明 */
  description: string;
  /** タスクの完了フラグ */
  isComplete: boolean;
  /** DeleteボタンのHandler */
  deleteTodoHandler: UseTodos['deleteTodo'];
  /** UpdateボタンのHandler */
  updateTodoHandler: UseTodos['updateTodo'];
}
