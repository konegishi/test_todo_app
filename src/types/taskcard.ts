import { Todo, UseTodos } from './todos';

/**
 * TaskCardコンポーネントのProps
 */
export interface TaskCardProps {
  /** タスクのリスト */
  todos: Todo[];
  /** DeleteボタンのHandler */
  deleteTodoHandler: UseTodos['deleteTodo'];
  /** UpdateボタンのHandler */
  updateTodoHandler: UseTodos['updateTodo'];
}
