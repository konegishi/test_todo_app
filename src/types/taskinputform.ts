import { User } from '@supabase/supabase-js';
import { UseTodos } from './todos';

/**
 * TaskInputFormコンポーネントのProps
 */
export interface TaskInputFormProps {
  /** ユーザー情報 */
  user: User;
  /** タスク名 */
  name?: string;
  /** AddボタンのHandler */
  addTodoHandler: UseTodos['addTodo'];
}
