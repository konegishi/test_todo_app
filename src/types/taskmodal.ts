import { Dispatch, SetStateAction } from 'react';
import { UseTodos } from './todos';

/**
 * TaskModalコンポーネントのProps
 */
export interface TaskModalProps {
  /** タスク名 */
  title: string;
  /** タスクのid */
  id: number;
  /** タスクの説明 */
  description: string;
  /** Modal開閉用メソッド */
  onClickHandler: Dispatch<SetStateAction<boolean>>;
  /** DeleteボタンのHandler */
  deleteTodoHandler: UseTodos['deleteTodo'];
  /** UpdateボタンのHandler */
  updateTodoHandler: UseTodos['updateTodo'];
}
