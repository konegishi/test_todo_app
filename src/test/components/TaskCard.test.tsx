import { render } from '@testing-library/react';
import React from 'react';
import TaskCard from '../../components/TaskCard';
import { TaskCardProps } from '../../types/taskcard';

describe('TaskCard component', () => {
  it('matches snapshot', () => {
    const dummyProps: TaskCardProps = {
      todos: [
        {
          task: '型定義を追加する',
          id: 10,
          description: 'types以下に配置する',
          isComplete: false,
        },
        {
          task: '型定義を追加する2',
          id: 11,
          description: 'types以下に配置する2',
          isComplete: false,
        },
      ],
      deleteTodoHandler: jest.fn(),
      updateTodoHandler: jest.fn(),
    };
    const { asFragment } = render(<TaskCard {...dummyProps} />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
