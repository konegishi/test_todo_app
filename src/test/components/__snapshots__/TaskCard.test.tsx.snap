// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskCard component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex-grow pt-4 px-4"
  >
    <div
      class=" bg-white rounded mb-4 shadow-lg pt-2 px-2"
    >
      <h2
        class="ml-2 mb-2 font-semibold text-2xl text-blueGray-800"
      >
        今日
      </h2>
      <div
        class="flex flex-row items-center border-t border-gray-200 py-2"
      >
        <input
          class="h-5 w-5 form-checkbox rounded-full mx-2 border-gray-300 focus:ring-0 focus:ring-opacity-50 text-blue-500"
          type="checkbox"
        />
        <span
          class="text-lg ml-2 truncate text-blueGray-800"
        >
          型定義を追加する
        </span>
        <button
          class="fas fa-ellipsis-v h-5 w-5 ml-auto mr-2 text-gray-400"
        />
      </div>
      <div
        class="flex flex-row items-center border-t border-gray-200 py-2"
      >
        <input
          class="h-5 w-5 form-checkbox rounded-full mx-2 border-gray-300 focus:ring-0 focus:ring-opacity-50 text-blue-500"
          type="checkbox"
        />
        <span
          class="text-lg ml-2 truncate text-blueGray-800"
        >
          型定義を追加する2
        </span>
        <button
          class="fas fa-ellipsis-v h-5 w-5 ml-auto mr-2 text-gray-400"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
